### ISSUE TITLE
🐛 [Bug] Search doesn't find user names with two characters

### ISSUE THREAD

**Issue by sushidave** [Human-Connection/Human-Connection#2800]
## :bug: Bugreport

Search doesn't find user names ("slugs") with two characters.

Example:
![hc-bug-seach-doesnt-find-slugs-less-than-two-characters-example-new](https://user-images.githubusercontent.com/17728384/72520890-b4d08d00-385a-11ea-9096-63e1bd0f9906.png)

### Steps to reproduce the behavior
1. In the search field, type in a user name of less than three characters. 
2. The search results shows "Nothing found".

### Expected behavior

The search results should show those users.

### Version & Environment
 Type: [any]      
 - OS: [any]      
 - Browser: [any] 
 - Version [any]  
 - Device: []  

### Additional context

If user names of two characters are permitted the search should allow for two characters, too. 
Alternative: Change user names policy i.e. mininum three characters instead of two.

Similar issue with hashtags #2635

**Comment by Mogge** [comment]
Until now, we do not trigger a search if the search string is less than three characters. If we do so, we will a lot of ugly side effects, as can be seen in #2799 
The only idea I have, to get this fixed in the case for users is, that you must search explicitly for users to skip the three characters minimum rule, This could be done by starting the search string with an '@' Symbol.
In this case, you would have to search for `@tj` to get an result.
What do you think about this solution?

**Comment by sushidave** [comment]
Title changed because user @mid (three characters) can't be mentioned too.

**Comment by sushidave** [comment]
@Mogge I like your suggestion. Will it find users with a one character user name such as @_ too?

**Comment by Mogge** [comment]
Soon we will have a search result page. This page can be reached by pressing enter in the search field. After pressing enter the search is triggered independently of the length of the input. This should fix the problem.

**Comment by sushidave** [comment]
@Mogge Thank you. Is the search result page going to replace the current limited seach result preview or is it additionally?

**Comment by Mogge** [comment]
It is additionally. When you press enter, you will see a page showing the results of the search more detailed as in the drop down. Also there are shown more results than just five for each type as it is now. The work on this page is in progressed, planned to be deployed in two weeks to production.

**Comment by sushidave** [comment]
@Mogge So the results on the preview and the first results of the views of the results page are identical (except that the results page contains more details)? In other words, no more limitation for the length of the input for the preview?

**Comment by Mogge** [comment]
No. If you press enter, the search triggers no matter how many characters are in the search box. When not pressing enter, the search triggers after having an input of at least 3 chars.
You can find user @tj now, when typing tj and press enter:
![image](https://user-images.githubusercontent.com/15882241/77760030-0608b400-7036-11ea-9c8e-792d5c28da6a.png)

**Comment by sushidave** [comment]
@Mogge Thanks. I tried to search for user "tj" and it worked. 

The search function seems to work now for usernames that start with letters but it's inconsistent with those containing numbers or special characters:

Examples:
- "_" → hitting Enter doesn't show the search result list
- "_michael" → _ to _mi + Enter doesn't trigger search result list, but listed with _mic and more
- "12c" → user not listed with 1 or 12, but listed with 12c
- "360" → user not listed
- "420" → not listed with 4 or 42, but listed with 420
- "4915255184321" → not listed with 4 to 491, but listed with 4915 and more
- "3lvis" → not listed with 3, 3l, 3lv, but listed with 3liv or 3livs
- "360fun" → not listed with 3 to 360, but listed with 360f to 360fun
- "911-gzm" → listed with 9 to 911-gzm
- "a_drian" → a to a_d + Enter doesn't trigger search result list, but listed with a_dr and more

Please re-check.

For the sake of a consistent user experience: How about not triggering the list automatically any longer and users need to press Enter in any case to get some search results?
